File: 2022/day1/Go/solution.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	input, err := os.ReadFile("../input.txt")
	if err != nil {
		panic(err)
	}

	// Split input into lines
	lines := strings.Split(string(input), "\n")
	solution1 := part1(lines)
	solution2 := part2(lines)

	fmt.Println("Solution 1:", solution1)
	fmt.Println("Solution 2:", solution2)
}

func part1(lines []string) int {
	return 0
}

func part2(lines []string) int {
	return 0
}

File: 2022/day1/solution.js
const fs = require('fs');

// Part 1
const part1 = () => {
    try {
        const fileContents = fs.readFileSync('../input.txt', 'utf8');
        const lines = fileContents.split('\n');

        let elfArray = [0];
        lines.map((line) => {
            if (line == '') {
                elfArray.push(0);
                return;
            }

            elfArray[elfArray.length - 1] += parseInt(line);
        });

        // Sort High to Lowest
        elfArray.sort((a, b) => b - a);
        console.log(elfArray[0]);
    } catch (error) {
        console.log(error);
    }
};

// Part 2
const part2 = () => {
    try {
        const fileContents = fs.readFileSync('../input.txt', 'utf8');
        const lines = fileContents.split('\n');

        let elfArray = [0];
        lines.map((line) => {
            if (line == '') {
                elfArray.push(0);
                return;
            }

            elfArray[elfArray.length - 1] += parseInt(line);
        });

        // Sort High to Lowest
        elfArray.sort((a, b) => b - a);
        console.log(elfArray[0] + elfArray[1] + elfArray[2]);
    } catch (error) {
        console.log(error);
    }
};

part1();
part2();
